spec/requests/api/v1/invoices/can_find_invoices_by_query_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+describe "user queries invoices with find" do
+  it "by the customer_id" do
+    invoice1 = Fabricate(:invoice)
+
+    get "/api/v1/invoices/find?customer_id=#{invoice1.customer_id}"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(invoice1.id)
+  end
+
+  it "by the merchant_id" do
+    invoice1 = Fabricate(:invoice)
+
+    get "/api/v1/invoices/find?merchant_id=#{invoice1.merchant_id}"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(invoice1.id)
+  end
+
+  it "by the status" do
+    invoice1 = Fabricate(:invoice)
+
+    get "/api/v1/invoices/find?status=#{invoice1.status}"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(invoice1.id)
+  end
+
+  it "by the invoice id" do
+    invoice1 = Fabricate(:invoice)
+
+    get "/api/v1/invoices/find?status=#{invoice1.status}"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(invoice1.id)
+  end
+
+  # it "can find by created_at" do
```

Comment:
Not positive what all you've tried, but my recommendation is to add the created at explicitly to the create in your test and then use that when you find. Let's talk about this tomorrow as well.